Documentation/CRDs/Cluster/external-cluster/external-cluster.md

```diff
@@ -305,3 +286,59 @@ you can export the settings from this cluster with the following steps.
 
 !!! important
     For other clusters to connect to storage in this cluster, Rook must be configured with a networking configuration that is accessible from other clusters. Most commonly this is done by enabling host networking in the CephCluster CR so the Ceph daemons will be addressable by their host IPs.
+
+## Upgrades
+
+1. If consumer cluster doesn't have restricted caps, this will upgrade all the default csi-users (non-restricted)
+
+    ```console
+    python3 create-external-cluster-resources.py --upgrade
+    ```
+
+2. If the consumer cluster has restricted caps
+
+    Restricted users created using `--restricted-auth-permission` flag need to pass mandatory flags: '`--rbd-data-pool-name`(if it is a rbd user), `--k8s-cluster-name` and `--run-as-user`' flags while upgrading, in case of cephfs users if you have passed `--cephfs-filesystem-name` flag while creating csi-users then while upgrading it will be mandatory too. In this example the user would be `client.csi-rbd-node-rookstorage-replicapool` (following the pattern `csi-user-clusterName-poolName`)
+
+    ```console
+    python3 create-external-cluster-resources.py --upgrade --rbd-data-pool-name replicapool --k8s-cluster-name rookstorage --run-as-user client.csi-rbd-node-rookstorage-replicapool
+    ```
+
+    !!! note
+        An existing non-restricted user cannot be converted to a restricted user by upgrading.
+        The upgrade flag should only be used to append new permissions to users. It shouldn't be used for changing a csi user already applied permissions. For example, you shouldn't change the pool(s) a user has access to.
+
+
+### Upgrade cluster to utilize new feature
+
+Some Rook upgrades may require re-running the import steps, or may introduce new external cluster features that can be most easily enabled by re-running the import steps.
+
+To re-run the import steps with new options, the python script should be re-run using the same configuration options that were used for past invocations, plus the configurations that are being added or modified.
+
+Starting with Rook v1.15, the script stores the configuration in the external-cluster-user-command configmap for easy future reference.
+
+* arg: Exact arguments that were used for for processing the script. Argument that are decided using the Priority: command-line-args > config.ini file values > default values.
+
+Example `external-cluster-user-command` ConfigMap:
+
+```console
+$ kubectl get cm  -nrook-ceph external-cluster-user-command  -o jsonpath='{.data.args}'
```

Comment:
Can we separate the command and the example? 

The command itself is **the** command users need to use to get the last-applied config, and I think readers will have the easiest time if that is explicitly stated. 

What comes after is an example of the output. TBH, I'm on the fence about whether to include example output or not. Users don't always follow directions well, and I have seen users just copy example output as though it were their own. In this case, I don't know if the example output adds much help for users, so let's omit the example output for now to try to avoid that. We can add some back in if any readers complain.

Once we do that, we can actually make a step-by-step guide something like this. I like the step-by-step flow when we can use it so that users know what to do clearly, and are therefore less likely to make mistakes or use tools in ways they aren't intended:

1. Get the last-applied config, if its available
   ```
   command
   ```
2. Copy the output to `config.ini`
3. Make any desired modifications and additions to `config.ini
4. Run the script again using the config file

!!! Note
    If the last-applied config is unavailable, run the current version of the script again using previously-applied config and CLI flags.